### PR TITLE
Resolve Can not export Coupon Code to CSV,XML issue29277

### LINKED ISSUE #29277: Can not export Coupon Code to CSV,XML
### Preconditions (*)

1. Magento 2.4-develop

### Steps to reproduce (*)

1. Go to backend, Marketing->Cart Price Rules
2. Create new rule, select auto generate coupon , generate coupon
3. In the grid, choose some coupon (must select by checkbox)
4. Click Export CSV or xml

image

### Expected result (*)

1. Export Successfully

### Actual result (*)

1. 404 not found
---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by edenduong**
### Description (*)

Resolve Can not export Coupon Code to CSV,XML

The problem is the controller action can not handle POST request (because it does not implement  POST Interface)

### Fixed Issues (if relevant)

1. Fixes magento/magento2#29277: Can not export Coupon Code to CSV, XML

### Manual testing scenarios (*)

1. Go to backend, Marketing->Cart Price Rules
2. Create new rule, select auto generate coupon , generate coupon
3. In the grid, choose some coupon (must select by checkbox)
4. Click Export CSV or XML

image

#### Expected result 

1. Export Successfully

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by lbajsarowicz** [comment]
@magento run all tests

**Comment by lbajsarowicz** [comment]
@edenduong Could you provide MFTF test for that scenario, please?

**Comment by edenduong** [comment]
> @edenduong Could you provide MFTF test for that scenario, please?

I will try to add Integration or MFTF test. THank you ! :)

**Comment by edenduong** [comment]
@lbajsarowicz : I have just cover my changes by Integration Test. Thanks !

**Comment by lbajsarowicz** [comment]
@magento run all tests

After getting the tests :heavy_check_mark: , I'll approve the PR.

**Comment by edenduong** [comment]
@magento run all tests

**Comment by gabrieldagama** [comment]
@magento run Functional Tests B2B

File: dev/tests/integration/testsuite/Magento/SalesRule/Controller/Adminhtml/Promo/Quote/ExportCoupons/ExportCouponsController.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+use Magento\Framework\App\ResourceConnection;
+use Magento\SalesRule\Model\ResourceModel\Rule\Collection as RuleCollection;
+use Magento\SalesRule\Model\Rule;
+use Magento\TestFramework\Helper\Bootstrap;
+use Magento\TestFramework\TestCase\AbstractBackendController;
+
+/**
+ * Abstract controller for test export coupon
+ */
+abstract class ExportCouponsController extends AbstractBackendController
```
Comment (sidolov) [inline dev/tests/integration/testsuite/Magento/SalesRule/Controller/Adminhtml/Promo/Quote/ExportCoupons/ExportCouponsController.php]: Please, avoid usage of abstract classes and inheritance

**Review by sidolov — CHANGES_REQUESTED** [review]

**Comment by engcom-Charlie** [comment]
Hi @edenduong can you please look through at requested changes https://github.com/magento/magento2/pull/29278#discussion_r473233310?
Thank you.

**Comment by engcom-Charlie** [comment]
Hi, @edenduong I'll try to continue with your PR.

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Review by sidolov — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sidolov, thank you for the review.
[ENGCOM-8153](https://jira.corp.magento.com/browse/ENGCOM-8153) has been created to process this Pull Request
:eight_spoked_asterisk: @sidolov, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Manual testing scenario:**

1. Go to backend, Marketing->Cart Price Rules;
2. Create new rule, select auto generate coupon , generate coupon;
3. In the grid, choose some coupon (must select by checkbox);
4. Click Export CSV or XML;

**Before :** :heavy_multiplication_x: 404 Error. Page not found.

Screenshot from 2020-09-08 09-36-40

**After:** :heavy_check_mark: Coupon codes were successfully exported

Peek 2020-09-08 10-03

**Pull request merged**